magma/circuit.py

```diff
@@ -392,12 +469,18 @@ def set_debug_info(self, debug_info):
         self.debug_info = debug_info
 
     def __str__(self):
-        if self.name:
-            return f"{self.name}<{type(self)}>"
-        name = f"AnonymousCircuitInst{id(self)}"
-        interface = ", ".join(f"{name}: {type(value)}"
-                              for name, value in self.interface.ports.items())
-        return f"{name}<{interface}>"
+        name = self.name
+        if not name:
+            name = f"AnonymousCircuitInst{id(self)}"
+        if type(self) is AnonymousCircuitType:
+            interface = ", ".join(
+                f"{name}: {type(value)}"
+                for name, value in self.interface.ports.items()
+            )
+            name += f"<{interface}>"
+        else:
+            name += f"<{type(self)}>"
+        return name
```

Comment:
This was an artifact for the updated tests since it improves the "stringification" logic for named anon circuit instances.